scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/signal/SignalApiImpl.java

```diff
@@ -108,96 +114,109 @@ public boolean isReceived(String signalName) throws SignalApiException {
+                signal.setOutputValues(signalParameters.get(signalName));
+                synchronization.remove(SIGNALS_CHANNEL + jobId, READY_PREFIX + signalName);
+                synchronization.put(SIGNALS_CHANNEL + jobId, signalName, signal);
             }
-            synchronization.compute(SIGNALS_CHANNEL, jobId, "{k, x -> " + actions.toString() + "x}");
-            return true;
         } catch (InvalidChannelException e) {
             throw new SignalApiException("Could not read signals channel", e);
-        } catch (CompilationException | IOException e) {
+        } catch (IOException e) {
             throw new SignalApiException("Could not add signals for the job " + jobId, e);
         }
     }
 
     @Override
-    public boolean removeSignal(String signalName) throws SignalApiException {
+    public void sendManySignals(Set<String> signalsSubSet) throws SignalApiException {
+        Map<String, Map<String, String>> signalParameters = new LinkedHashMap<>();
+        signalsSubSet.forEach(signalName -> signalParameters.put(signalName, new HashMap<>()));
+        sendManySignals(signalParameters);
+    }
+
+    @Override
+    public void removeSignal(String signalName) throws SignalApiException {
         try {
             init();
-            synchronization.compute(SIGNALS_CHANNEL, jobId, "{k, x -> x.remove('" + signalName + "');x}");
-            return true;
+            synchronization.remove(SIGNALS_CHANNEL + jobId, signalName);
         } catch (InvalidChannelException e) {
             throw new SignalApiException("Could not read signals channel", e);
-        } catch (CompilationException | IOException e) {
+        } catch (IOException e) {
             throw new SignalApiException("Could not remove signal for the job " + jobId, e);
         }
     }
 
     @Override
-    public boolean removeManySignals(Set<String> signalsSubSet) throws SignalApiException {
+    public void removeManySignals(Set<String> signalsSubSet) throws SignalApiException {
         try {
             init();
-            StringBuilder actions = new StringBuilder();
             for (String signal : signalsSubSet) {
-                actions.append(" x.remove('" + signal + "'); ");
+                synchronization.remove(SIGNALS_CHANNEL + jobId, signal);
             }
-            synchronization.compute(SIGNALS_CHANNEL, jobId, "{k, x -> " + actions.toString() + "x}");
-            return true;
         } catch (InvalidChannelException e) {
             throw new SignalApiException("Could not read signals channel", e);
-        } catch (CompilationException | IOException e) {
+        } catch (IOException e) {
             throw new SignalApiException("Could not add ready signals for the job " + jobId, e);
         }
     }
 
     @Override
-    public Set<String> getJobSignals() throws SignalApiException {
+    public Map<String, Signal> getJobSignals() throws SignalApiException {
```

Comment:
as the getJobSignals return type has changed (and also checkForSignals)

Could you search for usage of these methods in the proactive-examples project and adapt them accordingly before merging ?

Reply:
yes, I will search